utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,172 @@
+"""
+Functions for actions pertaining to history files.
+"""
+
+from CIME.XML.standard_module_setup import *
+from CIME.XML.component import Component
+
+import logging, glob, os, shutil, re
+
+def _iter_model_file_substrs():
+    drv_comp = Component()
+    models = drv_comp.get_valid_model_components()
+
+    for model in models:
+        model = "cpl" if model == "DRV" else model.lower()
+        yield model
+
+def _get_all_hist_files(testcase, model, suffix="", from_dir=os.getcwd()):
+    suffix = (".%s" % suffix) if suffix else ""
+    test_hists = glob.glob("%s/%s.%s*.h?.nc%s" % (from_dir, testcase, model, suffix))
+    test_hists.extend(glob.glob("%s/%s.%s*.h.nc%s" % (from_dir, testcase, model, suffix)))
+    test_hists.extend(glob.glob("%s/%s.%s*.h?.*.nc%s" % (from_dir, testcase, model, suffix)))
+    test_hists.extend(glob.glob("%s/%s.%s*.h.*.nc%s" % (from_dir, testcase, model, suffix)))
+    if suffix == "":
+        test_hists.extend(glob.glob("%s/%s.h.nc" % (from_dir, model)))
+        test_hists.extend(glob.glob("%s/%s.h?.nc" % (from_dir, model)))
+    test_hists.sort()
+    return test_hists
+
+def move(case, suffix):
+    """
+    Change the suffix for the most recent batch of hist files
+    """
+    rundir   = case.get_value("RUNDIR")
+    testcase = case.get_value("CASE")
+
+    # Loop over models
+    comments = "Moving hist files to suffix '%s'\n" % suffix
+    num_moved = 0
+    for model in _iter_model_file_substrs():
+        comments += "  Moving hist files for model '%s'\n" % model
+        test_hists = _get_all_hist_files(testcase, model, from_dir=rundir)
+        num_moved += len(test_hists)
+        for test_hist in test_hists:
+            new_file = "%s.%s" % (test_hist, suffix)
+            if os.path.exists(new_file):
+                os.remove(new_file)
+
+            comments += "    Copying '%s' to '%s'\n" % (test_hist, new_file)
+            shutil.copy(test_hist, new_file)
+
+    expect(num_moved > 0, "move failed: no hist files found in rundir '%s'" % rundir)
+
+    return comments
+
+def _compare_hists(case, suffix1="", suffix2="", from_dir1=os.getcwd(), from_dir2=os.getcwd()):
+
+    testcase = case.get_value("CASE")
```

Comment:
This is the most important function in the file. It encapsulates comparing a sets of hists files which is used by compare_test and compare_baseline.